gaps/gap-25_golem_certificates/gap-25_golem_certificates.md

```diff
@@ -0,0 +1,181 @@
+- Signing legal Agreements with partner companies developing software on Golem, in which they commit not to do anything malicious
+- Distribute permissions for signing Payloads to external companies taking responsibility for Payloads safety
+
+When designing certificates structure, that will allow to meet requirements mentioned above, we must keep in mind future use cases, that might require signing with certificates:
+
+- KYC
+- Inbound network traffic
+
+That means, that certificates will be used to sign different kinds of permissions. In case of transferring certification to other companies or signing legal Agreements, we need to ensure, that we keep control, which permissions they will be able to use.
+
+In the future of Golem Network we expect, that new independent features will be implemented by community and they may need to use certification. This document aims to standardize this approach.
+
+## Specification
+
+We decided to use [x509 arbitrary extensions](https://www.openssl.org/docs/man1.1.1/man5/x509v3_config.html) to enrich certificates with custom fields describing permissions.
+
+### Custom OIDs
+
+To use arbitrary extensions we need to apply to https://pen.iana.org/pen/PenApplication.page to assign OID (Private Enterprise Number) for GolemFactory.
+In further documentation I assume we got number `1.3.6.1.4.1.60000`, until we will acquire real OID.
+
+Having our OID, we can define arbitrary semantic to numbers in our namespace:
+
+
+| Name                       | OID                      | Semantic                                                             |
+| -------------------------- | ------------------------ | -------------------------------------------------------------------- |
+| GolemFactory               | 1.3.6.1.4.1.60000 (TODO) | Root OID for Golem Factory                                           |
+| GolemCertificates          | ${GolemFactory}.1        | Root for Golem certificates parameters                               |
+| GolemCertificatesAllowance | ${GolemCertificates}.1   | List of permissions to sign certain Golem features using certificate |
+
+Example values of `GolemCertificatesAllowance` field (this list will be used in further examples, to illustrate how certification works):
+
+
+| Field             | Semantic                                                                        |
+| ----------------- | ------------------------------------------------------------------------------- |
+| all               | Certificate is allowed to sign any feature (even not existing future use cases) |
+| manifest-outbound | Allowed to sign manifests with outbound network enabled                         |
+| inbound           | Allowed to sign inbound allowance for Requestors.                               |
+| ...               | {List is non-exhaustive - new features will be created in future}               |
+
```

Comment:
How do we manage/maintain the list of allowed `GolemCertificatesAllowance` values? This calls for a public, 'official' list, maybe maintained as part of Golem Standards (here: https://github.com/golemfactory/golem-architecture/tree/master/standards ?)

Reply:
I wrote somewhere, that we will update list in this GAP, but maybe it is good idea to maintain it in standards. But we must take care that we have links to standards everywhere, because sometimes I was having hard time to find something.